src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -0,0 +1,220 @@
+                {
+                    var declarationFile = await metadataAsSourceFileService.GetGeneratedFileAsync(
+                        document.Project, symbol, allowDecompilation: false, cancellationToken).ConfigureAwait(false);
+
+                    var linePosSpan = declarationFile.IdentifierLocation.GetLineSpan().Span;
+                    location = new LSP.Location
+                    {
+                        Uri = new Uri(declarationFile.FilePath),
+                        Range = ProtocolConversions.LinePositionToRange(linePosSpan),
+                    };
+                }
+            }
+            else
+            {
+                location = await ProtocolConversions.DocumentSpanToLocationAsync(documentSpan, cancellationToken).ConfigureAwait(false);
+            }
+
+            // TO-DO: The Origin property should be added once Rich-Nav is completed.
+            // https://github.com/dotnet/roslyn/issues/42847
+            var result = new LSP.VSReferenceItem
+            {
+                ContainingMember = properties.TryGetValue(
+                    AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var referenceContainingMember) ? referenceContainingMember : null,
+                ContainingType = properties.TryGetValue(
+                    AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var referenceContainingType) ? referenceContainingType : null,
+                DefinitionId = definitionId,
+                DefinitionText = definitionText,    // Only definitions should have a non-null DefinitionText
+                DisplayPath = location?.Uri.LocalPath,
+                DocumentName = documentSpan == default ? null : documentSpan.Document.Name,
+                Id = id,
+                Kind = symbolUsageInfo.HasValue ? ProtocolConversions.SymbolUsageInfoToReferenceKinds(symbolUsageInfo.Value) : new ReferenceKind[] { },
+                Location = location,
+                ProjectName = documentSpan == default ? null : documentSpan.Document.Project.Name,
+                ResolutionStatus = ResolutionStatusKind.ConfirmedAsReference,
+            };
+
+            // Properly assigning the text property.
+            if (id == definitionId)
+            {
+                result.Text = definitionText;
+            }
+            else if (documentSpan != default)
+            {
+                var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(
+                    documentSpan, cancellationToken).ConfigureAwait(false);
+                var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+                var docText = await documentSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                result.Text = new ClassifiedTextElement(
+                    classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+            }
```

Comment:
as above, this is very confusing.  This could simply be:

```c#
var text = await ComputeTextAsync(...);
```

then just initialize the object with that value.